File: app/api/schemas/req/collector.go
package req

// Serial 本地串口，主要指两个485串口
type Serial struct {
	Name       string `json:"name"`       // 串口名称
	DeviceName string `json:"deviceName"` // 设备名称
	BaudRate   int    `json:"baudRate"`   // 波特率
	DataBit    int    `json:"dataBit"`    // 数据位
	StopBit    string `json:"stopBit"`    // 停止位
	Check      string `json:"check"`      // 检验
}

// TcpClient 南向网口，主要指接入网桥的设备，即485转tcp协议
type TcpClient struct {
	Name string `json:"name"` // 名称
	Ip   string `json:"ip"`   // Ip地址
	Port int    `json:"port"` // 端口号
}

// TcpServer 北向网口，主要指北向接入的设备，即北向接入的tcp协议
type TcpServer struct {
	Name string `json:"name"`
	Port int    `json:"port"`
}

type Mqtt struct {
	Name string `json:"name"` // 名称
}

type Channel struct {
	Name string `json:"name"` // 名称
}

type CollectorSaveReq struct {
	Name      string    `json:"name" form:"name" binding:"required"`
	Type      string    `json:"type" form:"type" binding:"required"`
	Serial    Serial    `json:"serial" form:"serial"`
	TcpClient TcpClient `json:"tcpClient" form:"tcpClient"`
	TcpServer TcpServer `json:"tcpServer" form:"tcpServer"`
	Mqtt      Mqtt      `json:"mqtt" form:"mqtt"`
	Channel   Channel   `json:"channel" form:"channel"`
	Timeout   int       `json:"timeout" form:"timeout" default:"3000"`
	Interval  int       `json:"interval" form:"interval" default:"200"`
}

File: frontend/src/service/api/collector.ts
import { request } from '../http'

// 串口配置
export interface Serial {
  name: string
  deviceName: string
  baudRate: number
  dataBit: number
  stopBit: string
  check: string
}

// TCP客户端配置
export interface TcpClient {
  name: string
  ip: string
  port: number
}

// TCP服务端配置
export interface TcpServer {
  name: string
  port: number
}

// MQTT配置
export interface Mqtt {
  name: string
}

// 通道配置
export interface Channel {
  name: string
}

// 4G GPRS配置
export interface FourGPRS {
  name: string
}

// 采集器接口
export interface CollectorInfo {
  name: string
  type: string
  address: string
  serial: Serial
  tcpClient: TcpClient
  tcpServer: TcpServer
  mqtt: Mqtt
  channel: Channel
  fourGPRS: FourGPRS
  timeout: number
  interval: number
  enable: boolean
  createdAt: number
  updatedAt: number
}

// 采集器创建/更新请求
export interface CollectorRequest {
  name: string
  type: string
  address: string
  serial: Serial
  tcpClient: TcpClient
  tcpServer: TcpServer
  mqtt: Mqtt
  channel: Channel
  fourGPRS: FourGPRS
  timeout: number
  interval: number
  enable: boolean
}

// 创建采集器
export function createCollector(data: CollectorRequest) {
  return request.Post<Service.ResponseResult<null>>('/collector', data)
}

// 更新采集器
export function updateCollector(data: CollectorRequest) {
  return request.Put<Service.ResponseResult<null>>('/collector', data)
}

// 删除采集器
export function deleteCollector(name: string) {
  return request.Delete<Service.ResponseResult<null>>(`/collector/${name}`)
}

// 获取采集器详情
export function getCollector(name: string) {
  return request.Get<Service.ResponseResult<CollectorInfo>>(`/collector/${name}`)
}

// 获取采集器列表
export function getCollectorList(params: { page: number; pageSize: number; type?: string; enable?: boolean }) {
  return request.Get<Service.ResponseResult<Service.PageData<CollectorInfo[]>>>('/collectors', { params })
}

// 获取采集器数量
export function getCollectorCount(params: { type?: string; enable?: boolean }) {
  return request.Get<Service.ResponseResult<{ count: number }>>('/collector/count', { params })
} 